splink/aws/aws_linker.py

```diff
@@ -0,0 +1,186 @@
+import sqlglot
+from splink.linker import Linker, SplinkDataFrame
+import logging
+from splink.logging_messages import execute_sql_logging_message_info, log_sql
+
+logger = logging.getLogger(__name__)
+
+# import utils for communicating with aws
+import awswrangler as wr
+import boto3
+from splink.aws.aws_utils import boto_utils
+
+
+class AWSDataFrame(SplinkDataFrame):
+    def __init__(self, templated_name, physical_name, aws_linker):
+        super().__init__(templated_name, physical_name)
+        self.aws_linker = aws_linker
+
+    @property
+    def columns(self):
+        t = self.get_schema_info(self.physical_name)
+        d = wr.catalog.get_table_types(
+            database=t[0],
+            table=t[1]
+        )
+
+        return list(d.keys())
+
+    def validate(self):
+        pass
+    
+    def drop_table_from_database(self, force_non_splink_table=False):
+        
+        self._check_drop_folder_created_by_splink(force_non_splink_table)
+        self._check_drop_table_created_by_splink(force_non_splink_table)
+        self.aws_linker.drop_table_from_database_if_exists(self.physical_name)
+        self.aws_linker.delete_table_from_s3(self.physical_name)
+    
+    def _check_drop_folder_created_by_splink(self, force_non_splink_table=False):
+        
+        filepath = self.aws_linker.boto_utils.s3_output
+        filepath = filepath.split("/")[-3:-1]
+        # validate that the write path is valid
+        valid_path = [
+            self.aws_linker.boto_utils.s3_output_name_prefix, 
+            self.aws_linker.boto_utils.session_id
+            ] == filepath
+        if not valid_path:
+            if not force_non_splink_table:
+                raise ValueError(
+                    f"You've asked to drop data housed under the filepath "
+                    f"{self.aws_linker.boto_utils.s3_output} "
+                    "from your s3 output bucket, which is not a folder created by Splink. "
+                    "If you really want to delete this data, you can do so by setting "
+                    "force_non_splink_table=True."
+                )
+        
+        # validate that the ctas_query_info is for the given table we're interacting with
+        if self.aws_linker.ctas_query_info[self.physical_name]["ctas_table"] != self.physical_name:
+            raise ValueError(
+                    f"The recorded metadata for {self.physical_name} that you're "
+                    "attempting to delete does not match the recorded metadata on s3. "
+                    "To prevent any tables becoming corrupted on s3, this run will be "
+                    "terminated. Please retry the link/dedupe job and report the issue "
+                    "if this error persists."
+                )
+
+    def as_record_dict(self, limit=None):
```

Comment:
In addition to the `as_record_dict()` method, we also need an `as_pandas_dataframe` method (which, in this case, is the same, except returns `out_df`, without needing the `to_dict(orient='records)`) part